File: 02_ProgrammingFundamentals/_05_Lists/_02_Exercise/_08_AnonymousThreat.java
package _02_ProgrammingFundamentals._05_Lists._02_Exercise;

import java.util.Arrays;
import java.util.List;
import java.util.Scanner;
import java.util.stream.Collectors;

public class _08_AnonymousThreat {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input values
        List<String> stringList = Arrays.stream(scan.nextLine().split("\\s+")).collect(Collectors.toList());
        String commandLine = scan.nextLine();

        // Read and execute commands based on the input
        while (!commandLine.equals("3:1")) {
            String[] commandsInformation = commandLine.split("\\s+");
            String command = commandsInformation[0];
            StringBuilder appendElements = new StringBuilder();

            if (command.equals("merge")) {
                int startIndex = Integer.parseInt(commandsInformation[1]);
                int endIndex = Integer.parseInt(commandsInformation[2]);

                if (startIndex <= stringList.size() - 1 && startIndex < 0 ) {
                    startIndex = 0;
                }

                if (endIndex >= 0 && endIndex > stringList.size() - 1) {
                    endIndex = stringList.size() - 1;
                }

                if (startIndex <= stringList.size() - 1 && endIndex >= 0) {
                    for (int i = startIndex; i <= endIndex; i++) {
                        appendElements.append(stringList.get(i));
                        stringList.remove(i);

                        i--;
                        endIndex--;
                    }

                    stringList.add(startIndex, appendElements.toString());
                }
            } else if (command.equals("divide")) {
                int index = Integer.parseInt(commandsInformation[1]);
                int partitions = Integer.parseInt(commandsInformation[2]);

                String toDivide = stringList.get(index);
                stringList.remove(index);

                int partitionSize = toDivide.length() / partitions;
                int begin = 0;

                for (int i = 1; i < partitions; i++) {
                    stringList.add(index, toDivide.substring(begin, begin + partitionSize));
                    index++;
                    begin += partitionSize;
                }

                stringList.add(index, toDivide.substring(begin));
            }

            commandLine = scan.nextLine();
        }

        // Print result
        System.out.println(String.join(" ", stringList));

        scan.close();
    }
}

File: 24_C#Fundamentals/_05_Lists/_02_Exercise/_08_AnonymousThreat/_08_AnonymousThreat.cs
using System;
using System.Text;

public class _08_AnonymousThreat
{
    public static void Main(string[] args)
    {
        // Read input values
        List<string> stringList = Console.ReadLine().Split().ToList();
        string commandLine = Console.ReadLine();

        // Read and execute commands based on the input
        while (!commandLine.Equals("3:1"))
        {
            string[] commandsInformation = commandLine.Split();
            string command = commandsInformation[0];
            StringBuilder appendElements = new StringBuilder();

            if (command.Equals("merge"))
            {
                int startIndex = int.Parse(commandsInformation[1]);
                int endIndex = int.Parse(commandsInformation[2]);

                if (startIndex > stringList.Count - 1 || startIndex < 0)
                {
                    startIndex = 0;
                }

                if (endIndex < 0 || endIndex > stringList.Count - 1)
                {
                    endIndex = stringList.Count - 1;
                }

                if (startIndex <= stringList.Count - 1 && endIndex >= 0)
                {
                    for (int i = startIndex; i <= endIndex; i++)
                    {
                        appendElements.Append(stringList[i]);
                    }

                    stringList.RemoveRange(startIndex, endIndex - startIndex + 1);
                    stringList.Insert(startIndex, appendElements.ToString());
                }
            }
            else if (command.Equals("divide"))
            {
                List<string> dividedInput = new List<string>();
                int index = int.Parse(commandsInformation[1]);
                int partitions = int.Parse(commandsInformation[2]);

                string word = stringList[index];
                int parts = word.Length / partitions;
                stringList.RemoveAt(index);

                for (int i = 0; i < partitions; i++)
                {
                    if (i == partitions - 1)
                    {
                        dividedInput.Add(word.Substring(i * parts));
                    }
                    else
                    {
                        dividedInput.Add(word.Substring(i * parts, parts));
                    }
                }

                stringList.InsertRange(index, dividedInput);
            }

            commandLine = Console.ReadLine();
        }

        // Print result
        Console.WriteLine(string.Join(" ", stringList));
    }
}
